src/AutoRest.CSharp/Mgmt/AutoRest/PostProcess/Internalizer.cs

```diff
@@ -157,6 +157,18 @@ private static bool HasDiscriminator(BaseTypeDeclarationSyntax node, [MaybeNullW
                         }
                     }
                 }
+                // When a base class with discriminator is only used as input parameter, it doesn't contain a deserialization method
+                // therefore adding a comment check here as a redundant verification
+                else if (classDeclaration.HasLeadingTrivia)
+                {
+                    var syntaxTriviaList = classDeclaration.GetLeadingTrivia();
+                    if (BuilderHelpers.DiscriminatorDescFixedPart.All(syntaxTriviaList.ToFullString().Contains))
+                    {
+                        var identifierNodes = syntaxTriviaList.FirstOrDefault().GetStructure()?.DescendantNodes().OfType<XmlCrefAttributeSyntax>();
+                        identifiers = identifierNodes.Select(identifier => identifier.Cref.ToFullString()).ToImmutableHashSet();
+                        return true;
+                    }
+                }
```

Comment:
If we guaranteed that all the class with a discriminator would have the leading trivia with some "cref"s, we could just use this as a criteria and do not need to check the deserialization method. This will simplify the code in a great deal.
There are some other cref might be in the xml document, do you think we need to verify the content of that trivia?

Reply:
I think it is guaranteed that all the classes with discriminator will have this trivia. And yes, the verification of the content is also needed, I will extract the cref only from the specified line.